modules/core/src/main/java/org/apache/ignite/internal/client/thin/ClientServicesImpl.java

```diff
@@ -119,8 +143,10 @@ private ServiceInvocationHandler(String name, long timeout, ClientClusterGroupIm
                 if (nodeIds != null && nodeIds.isEmpty())
                     throw new ClientException("Cluster group is empty.");
 
-                return ch.service(ClientOperation.SERVICE_INVOKE,
-                    req -> writeServiceInvokeRequest(req, nodeIds, method, args),
+                boolean withCallCtx = callCtx != null;
+
+                return ch.service(withCallCtx ? ClientOperation.SERVICE_INVOKE_CALLCTX : ClientOperation.SERVICE_INVOKE,
```

Comment:
Lets' extend `ClientOperation.SERVICE_INVOKE`, it's redundant to introduce a new client operation type.

Reply:
Done, thanks!